lib/intercept.js

```diff
@@ -50,13 +50,18 @@ let allowNetConnect
  * @example
  * // Enables real requests for url that matches google and amazon
  * nock.enableNetConnect(/(google|amazon)/);
+ * @example
+ * // Enables real requests for url that includes google
+ * nock.enableNetConnect(host => host.includes('google'));
  */
 function enableNetConnect(matcher) {
   // TODO-12.x: Replace with `typeof matcher === 'string'`.
   if (_.isString(matcher)) {
     allowNetConnect = new RegExp(matcher)
   } else if (matcher instanceof RegExp) {
     allowNetConnect = matcher
+  } else if (matcher instanceof Function) {
```

Comment:
Because of https://github.com/nock/nock/pull/1754#issuecomment-571531407 this should use `typeof`.